ifrs17/Import/ImportStorage.ipynb

```diff
@@ -224,53 +224,52 @@
+                "\n                                    EqualityComparer<IfrsVariable>.Instance);",
                 "\n            ",
+                "\n            await querySource.Partition.SetAsync<PartitionByReportingNodeAndPeriod>(TargetPartition);",
+                "\n",
                 "\n            // TODO: print error if ",
                 "\n            //openingRawVariables.Select(x => x.DataNode).ToHashSet() != dataNodesWithPreviousPeriod",
                 "\n        }",
                 "\n        ",
-                "\n        //SetPartition to current Period",
-                "\n        await querySource.Partition.SetAsync<PartitionByReportingNodeAndPeriod>(TargetPartition);",
-                "\n",
-                "\n        var primaryScopeAtInceptionYear = allImportScopesAtInceptionYear.Where(dn => primaryScope.Contains(dn));",
-                "\n        var secondaryScopeAtInceptionYear = allImportScopesAtInceptionYear.Where(dn => secondaryScope.Contains(dn));        ",
-                "\n        if(allImportScopesAtInceptionYear.Any() && ImportFormat != ImportFormats.Opening) ",
-                "\n            openingIfrsVariables = openingIfrsVariables.Concat(await querySource.Query<IfrsVariable>()",
-                "\n                                                                .Where(iv => iv.Partition == TargetPartition && iv.AocType == AocTypes.BOP && iv.Novelty == Novelties.I)",
-                "\n                                                                .Where(iv => primaryScopeAtInceptionYear.Contains(iv.DataNode) ",
-                "\n                                                                             && EstimateTypesByImportFormat[InputSource.Opening.ToString()].Contains(iv.EstimateType) ",
-                "\n                                                                             || secondaryScopeAtInceptionYear.Contains(iv.DataNode))",
-                "\n                                                                .ToArrayAsync());",
-                "\n        ",
                 "\n        //Variables",
                 "\n        var rawVariables = parsedRawVariables.Concat(openingRawVariables)",
                 "\n                                             .Concat(await querySource.Query<RawVariable>()",
                 "\n                                                                      .Where(rv => rv.Partition == TargetPartition)",
                 "\n                                                                      .Where(rv => primaryScopeFromLinkedReinsurance.Contains(rv.DataNode)).ToArrayAsync()); ",
                 "\n        ",
-                "\n        var ifrsVariables = parsedIfrsVariables.Concat(openingIfrsVariables)",
+                "\n        var ifrsVariables = parsedIfrsVariables.Concat(openingIfrsVariables)  // TODO: ask Davide, why not union?",
```

Comment:
this concat can be an issue in the case of a partial (only some DataNode) reimport of opening.

Reply:
Using Union would solve most of the cases. The only exception is a **RE**import of opening where the Opening for CSM has been deleted and one for LC (same DataNode) added.

Reply:
Or is this case covered by the ImportFormat != ImportFormats.Opening condition of the if statement?
=> I think we need a clear test for this use case anyway (@nnikolopoulos )

Reply:
I can add a Case V in issue #140 where the scenario:

**RE**import of opening where the Opening for CSM has been deleted and one for LC (same DataNode) added.

is examined.

Reply:
ok, I checked the logic thoroughly and I am now convinced that in the case of a reimport of opening, the old opening are not loaded because of the ImportFormat != ImportFormats.Opening condition of the if statement.

This means that a re-import of openings is not incremental, i.e. every time the full set of openings needs to be provided. 

Is this the correct expected behavior?

Reply:
After our meeting, this is fixed: for data nodes at inception year, the secondary scope is loaded. 

This is realized by having moved the ImportFormat != ImportFormats.Opening condition from the if statement to the query logic for the primary scope. 

The test 3.6 in the Import Storage checks this behavior. Please resolve this conversation if everything is OK.